networking_ccloud/ml2/agent/common/switch.py

```diff
@@ -26,20 +27,59 @@
 LOG = logging.getLogger(__name__)
 
 
-def run_in_executor(type_):
+class FullSyncScheduled(Exception):
+    """Raised if a scheduled write-attempt was voluntarily interrupted because there's a full-sync in the queue
+
+    Callers catching this exception should wait for the attached future
+    attribute instead to know when their task finished.
+    """
+    def __init__(self, *args, future, **kwargs):
+        super().__init__(self, *args, **kwargs)
+        self.future = future
+
+
+def run_in_executor(type_, replacable_by_full_sync=False):
     """Decorator to run a method in the ThreadPoolExecutor of the class
 
     The wrapped method thus returns a futurist.Future now instead of its actual
     result.
+
+    If replacable_by_full_sync is set, the decorated method will not run if
+    there's a full-sync scheduled. Instead, it will raise FullSyncScheduled.
+    Additionally, the decorated method will also not get scheduled, if there's
+    full-sync scheduled already. In that case, the wrapped method returns the
+    futurist.Future of the scheduled full-sync instead of of its own. Callers
```

Comment:
Actually ... this could be fine ;) "instead of the futurist.Future of its own task" with "the futurist.Future" left out, because it's implied. rephrased a little.